kratos/elements/distance_calculation_flux_based_element.cpp

```diff
@@ -0,0 +1,502 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//			 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez, Pablo Becker
+//
+#include "includes/kratos_flags.h"
+#include "distance_calculation_flux_based_element.h"
+#include "includes/checks.h"
+
+
+namespace Kratos
+{
+
+/////specifications must be on top
+
+//gauss points for the 3D4N Tetra
+template <>
+void DistanceCalculationFluxBasedElement<3,4>::GetSimplexShapeFunctionsOnGauss(BoundedMatrix<double,4, 4>& Ncontainer)
+{
+    Ncontainer(0,0) = 0.58541020; Ncontainer(0,1) = 0.13819660; Ncontainer(0,2) = 0.13819660; Ncontainer(0,3) = 0.13819660;
+    Ncontainer(1,0) = 0.13819660; Ncontainer(1,1) = 0.58541020; Ncontainer(1,2) = 0.13819660; Ncontainer(1,3) = 0.13819660;
+    Ncontainer(2,0) = 0.13819660; Ncontainer(2,1) = 0.13819660; Ncontainer(2,2) = 0.58541020; Ncontainer(2,3) = 0.13819660;
+    Ncontainer(3,0) = 0.13819660; Ncontainer(3,1) = 0.13819660; Ncontainer(3,2) = 0.13819660; Ncontainer(3,3) = 0.58541020;
+}
+
+//gauss points for the 2D3N triangle
+template <>
+void DistanceCalculationFluxBasedElement<2,3>::GetSimplexShapeFunctionsOnGauss(BoundedMatrix<double,3,3>& Ncontainer)
+{
+    const double one_sixt = 1.0/6.0;
+    const double two_third = 2.0/3.0;
+    Ncontainer(0,0) = one_sixt; Ncontainer(0,1) = one_sixt; Ncontainer(0,2) = two_third;
+    Ncontainer(1,0) = one_sixt; Ncontainer(1,1) = two_third; Ncontainer(1,2) = one_sixt;
+    Ncontainer(2,0) = two_third; Ncontainer(2,1) = one_sixt; Ncontainer(2,2) = one_sixt;
+}
+
+//unused, but needed for compilation
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim,TNumNodes>::GetSimplexShapeFunctionsOnGauss(BoundedMatrix<double,TNumNodes,TNumNodes>& Ncontainer)
+{
+    KRATOS_ERROR << "CALLING TEMPLATED GetSimplexShapeFunctionsOnGauss" << std::endl;
+}
```

Comment:
```suggestion
```
This is required because in the bottom you're doing the specialization for 3 and 8 template arguments. Indeed, as it is right now it doesn't work as calling the 3 and 8 class specialization would get here and throw an error. You either remove that specialization or implement the `GetSimplexShapeFunctionsOnGauss` for the 3 and 8 template arguments case.

Reply:
OK. Now I see... For the hexas you're using the generic low performant function below...

Reply:
ok